cmd/githubrunnerscalesetlistener/main.go

```diff
@@ -67,17 +74,98 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err := run(rc, logger); err != nil {
-		logger.Error(err, "Run error")
+	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer stop()
+
+	g, ctx := errgroup.WithContext(ctx)
+
+	g.Go(func() error {
+		opts := runOptions{
+			serviceOptions: []func(*Service){
+				WithLogger(logger),
+			},
+		}
+		opts.serviceOptions = append(opts.serviceOptions, WithPrometheusMetrics(rc))
+
+		return run(ctx, rc, logger, opts)
+	})
+
+	if len(rc.MetricsAddr) != 0 {
+		g.Go(func() error {
+			metricsServer := metricsServer{
+				rc:     rc,
+				logger: logger,
+			}
+			g.Go(func() error {
+				<-ctx.Done()
+				return metricsServer.shutdown()
+			})
+			return metricsServer.run()
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		logger.Error(err, "Error encountered")
```

Comment:
Will we get error during SIGINT?

Reply:
No, unless I made a mistake :smile: 

The error will be returned in case g.Go callback returns an error. The SIGINT will cancel the context, and if we return `ctx.Err()`, then the g.Wait will return the error back causing `os.Exit(1)`. Otherwise, if callback stops on context cancellation and returns nil, this will exit with exit code 0.